improver/metadata/probabilistic.py

```diff
@@ -218,3 +218,45 @@ def find_percentile_coordinate(cube):
         raise ValueError(msg)
 
     return perc_coord
+
+
+def format_cell_methods_for_probability(cube, threshold_name):
+    """Update cell methods on a diagnostic cube to reflect the fact that the
+    data to which they now refer is on a coordinate.  Modifies cube in place.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube to update
+        threshold_name (str):
+            Name of the threshold-type coordinate to which the cell
+            method now refers
+    """
+    cell_methods = []
+    for cell_method in cube.cell_methods:
+        new_cell_method = iris.coords.CellMethod(
+            cell_method.method,
+            coords=cell_method.coord_names,
+            intervals=cell_method.intervals,
+            comments=f"of {threshold_name}",
+        )
+        cell_methods.append(new_cell_method)
+    cube.cell_methods = cell_methods
+
+
+def format_cell_methods_for_diagnostic(cube):
+    """Remove reference to threshold-type coordinate from cell methods on a
```

Comment:
It might be worth adding the word `comments` here, just so a future user understands that the significant part of this function is that it doesn't include a `comments` argument when the new cell method is created. Therefore perhaps something like this: 
```suggestion
    """Remove reference to threshold-type coordinate from cell method comments on a
```